src/goto-symex/symex_main.cpp

```diff
@@ -313,6 +373,12 @@ void goto_symext::symex_assume()
   dereference(cond, dereferencet::READ);
   replace_dynamic_allocation(cond);
 
+  if(options.get_bool_option("smt-symex-assume"))
+  {
+    if(check_incremental(cond, false, std::string()))
+      // incremental verification has succeeded
+      return;
+  }
```

Comment:
Is this correct? We should return when the early if the assume is true, but `check_incremental` also returns true when cond evaluates to false.

You'll probably need to replace the return value with a threeval object or an optional.

Reply:
The method `check_incremental` already adds `assume(false)`.